rakelib/utils.rake

```diff
@@ -0,0 +1,58 @@
+# run a command, pipe output through 'xcpretty' and store the output in CI artifacts
+def xcpretty(cmd, task, subtask = '')
+  name = (task.name + (subtask.empty? ? '' : "_#{subtask}")).gsub(/[:-]/, "_")
+  command = [*cmd].join(' && ')
+  xcpretty = `which xcpretty`
+
+  if ENV['CI']
+    sh "set -o pipefail && (#{command}) | tee \"#{ENV['CIRCLE_ARTIFACTS']}/#{name}_raw.log\" | xcpretty --color --report junit --output \"#{ENV['CIRCLE_TEST_REPORTS']}/xcode/#{name}.xml\""
+  elsif xcpretty && $?.success?
+    sh "set -o pipefail && (#{command}) | xcpretty -c"
+  else
+    sh command
+  end
+end
+
+# run a command and store the output in CI artifacts
+def plain(cmd, task, subtask = '')
```

Comment:
Even better would be to auto detect this similarly to how I detect the need for xcpretty. If possible I'd like to keep the other commands but mark them as private or something? Better to have shorter/simpler functions instead of one big one prone to breaking